src/gui/transferlistsortmodel.cpp

```diff
@@ -242,6 +242,9 @@ int TransferListSortModel::compare(const QModelIndex &left, const QModelIndex &r
             return threeWayCompare(totalL, totalR);
         }
 
+    case TransferListModel::TR_PERCENT_SELECTED:
+        return customCompare(leftValue.toFloat(), rightValue.toFloat());
+
```

Comment:
```suggestion
```
and put `case TransferListModel::TR_PERCENT_SELECTED:` just above `case TransferListModel::TR_POPULARITY:`.